sdk/cognitiveservices/azopenai/models.go

```diff
@@ -0,0 +1,209 @@
+	// will generate as if from the beginning of a new document. Maximum allowed size of string list is 2048.
+	Prompt any
+
+	// Up to 4 sequences where the API will stop generating further tokens. The returned text will not contain the stop sequence.
+	Stop any
+
+	// Whether to stream back partial progress. If set, tokens will be sent as data-only server-sent events as they become available,
+	// with the stream terminated by a data: [DONE] message.
+	Stream *bool
+
+	// The suffix that comes after a completion of inserted text.
+	Suffix *string
+
+	// What sampling temperature to use. Higher values means the model will take more risks. Try 0.9 for more creative applications,
+	// and 0 (argmax sampling) for ones with a well-defined answer. We generally
+	// recommend altering this or top_p but not both.
+	Temperature *float32
+
+	// An alternative to sampling with temperature, called nucleus sampling, where the model considers the results of the tokens
+	// with top_p probability mass. So 0.1 means only the tokens comprising the top
+	// 10% probability mass are considered. We generally recommend altering this or temperature but not both.
+	TopP *float32
+
+	// A unique identifier representing your end-user, which can help monitoring and detecting abuse
+	User *string
+}
+
+type CompletionResponse struct {
+	// REQUIRED
+	Choices []*Post200ApplicationJSONPropertiesItemsItem
+
+	// REQUIRED
+	Created *int32
+
+	// REQUIRED
+	ID *string
+
+	// REQUIRED
+	Model *string
+
+	// REQUIRED
+	Object             *string
+	CompletionResponse *Paths1Phq6SvDeploymentsDeploymentIDCompletionsPostResponses200ContentApplicationJSONSchemaPropertiesUsage
+}
+
+type EmbeddingRequest struct {
+	// REQUIRED; Input text to get embeddings for, encoded as a string. To get embeddings for multiple inputs in a single request,
+	// pass an array of strings. Each input must not exceed 2048 tokens in length. Unless you
+	// are embedding code, we suggest replacing newlines (\n) in your input with a single space, as we have observed inferior
+	// results when newlines are present.
+	Input any
+
+	// OPTIONAL; Contains additional key/value pairs not defined in the schema.
+	AdditionalProperties map[string]any
+
+	// input type of embedding search to use
+	InputType *string
+
+	// ID of the model to use. You can use the ModelsList operation to see all of your available models, or see our ModelsGet
+	// overview for descriptions of them.
+	Model *string
+
+	// A unique identifier representing your end-user, which can help monitoring and detecting abuse.
+	User *string
+}
+
+type EmbeddingResponse struct {
+	// REQUIRED
+	Data []*Paths1Xmf2L5DeploymentsDeploymentIDEmbeddingsPostResponses200ContentApplicationJSONSchemaPropertiesDataItems
+
+	// REQUIRED
+	Model *string
+
+	// REQUIRED
+	Object *string
+
+	// REQUIRED
+	Usage *Paths18Tiy9VDeploymentsDeploymentIDEmbeddingsPostResponses200ContentApplicationJSONSchemaPropertiesUsage
+}
+
+type ErrorResponse struct {
```

Comment:
Looks like the ErrorResponse struct is unused. This has happened in other modules and we usually just remove it with an autorest transform